src/main/plugin/iso19139.ca.HNAP/layout/tpl-csv.xsl

```diff
@@ -0,0 +1,211 @@
+      </referencesystem>
+
+      <!-- For each date grouped by the date type mapped to a readable value -->
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:citation/*/gmd:date">
+        <xsl:variable name="dateTypeCode" select="*/gmd:dateType/*/@codeListValue"/>
+        <xsl:variable name="dateTypeCodeReadable" select="tokenize($codelists/codelist[@name = 'gmd:CI_DateTypeCode']/entry[code/text() = $dateTypeCode]/value/text(), ';')[1]"/>
+
+        <xsl:element name="date-{$dateTypeCodeReadable}">
+          <xsl:value-of select="*/gmd:date/*/text()"/>
+        </xsl:element>
+      </xsl:for-each>
+
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:graphicOverview/*/gmd:fileName">
+        <image>
+          <xsl:value-of select="*/text()"/>
+        </image>
+      </xsl:for-each>
+
+
+      <!-- All keywords not having thesaurus reference or an empty thesaurusName -->
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:descriptiveKeywords/gmd:MD_Keywords[not(gmd:thesaurusName) or gmd:thesaurusName/*/gmd:title/(gco:CharacterString|gmx:Anchor)/text() = '']">
+        <xsl:variable name="keywordTypeCode" select="gmd:type/*/@codeListValue"/>
+        <xsl:variable name="keywordTypeCodeReadable" select="tokenize($codelists/codelist[@name = 'gmd:MD_KeywordTypeCode']/entry[code/text() = $keywordTypeCode]/value/text(), ';')[1]"/>
+
+        <xsl:for-each select="gmd:keyword[not(@gco:nilReason) or */text() != '']">
+          <xsl:choose>
+            <!-- All keywords with a valid keywordTypeCode -->
+            <xsl:when test="$keywordTypeCodeReadable != ''">
+              <xsl:element name="keyword-{$keywordTypeCodeReadable}">
+                <xsl:apply-templates mode="localised" select=".">
+                  <xsl:with-param name="langId" select="$langId"/>
+                </xsl:apply-templates>
+              </xsl:element>
+            </xsl:when>
+
+            <!-- All keywords without a valid keywordTypeCode -->
+            <xsl:otherwise>
+              <keyword-other>
+                <xsl:apply-templates mode="localised" select=".">
+                  <xsl:with-param name="langId" select="$langId"/>
+                </xsl:apply-templates>
+              </keyword-other>
+            </xsl:otherwise>
+          </xsl:choose>
+        </xsl:for-each>
+      </xsl:for-each>
+
+      <!-- All keywords with a valid thesaurus name -->
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:descriptiveKeywords/gmd:MD_Keywords[gmd:thesaurusName/*/gmd:title/(gco:CharacterString|gmx:Anchor)/text() != '']">
+        <xsl:variable name="thesaurusName">
+          <xsl:variable name="translationLanguage" select="gmd:thesaurusName/*/gmd:title/*/*/gmd:LocalisedCharacterString/@locale"/>
+          <xsl:choose>
+            <xsl:when test="$translationLanguage = '#eng'">
```

Comment:
This is incorrect.  If the main language of the document is French then English will be in `gmd:LocalisedCharacterString` and French will be in `gco:CharacterString`
But if the main language of the document is English then it will be reversed - English will be in `gco:CharacterString` and French will be in `gmd:LocalisedCharacterString`

There are other xsl files that correctly extract the values based on the document language - you may want to consider looking into those.